### ISSUE TITLE
PHPStan checking status, but no output

### ISSUE THREAD

**Issue by wallacio** [SanderRonde/phpstan-vscode#45]
TLDR; make sure the extension has access to a valid PHP executable, and is executing phpstan in a way that makes sense to VSCode.

It's taken me a while to work out what's going wrong with my local environment, but for the benefit of future me and others:

**Current config/use:**

Extension version: pre-release (3.0.1?). The 'current' release didn't work at all - PHPStan binary couldn't be found (which is essentially the source of the problem).

Using VSCode on W11. The code repository is managed by Composer and phpstan/phpstan installed as require-dev in the project. The dev environment runs in Homestead, and a native PHP8 binary is available to VSCode in Windows to keep VSCode happy with linting etc. (I'm using Intelephense quite happily). `composer install` is run from a shell in Homestead, so that paths/permissions work.

I can run phpstan happily via command in either WSL or Homestead with `vendor/bin/phpstan analyse`. There is a `phpstan.neon` in the project root. The phpstan scripts (phpstan and phpstan.phar) in vendor/bin are PHP (rather than binaries).

VSCode loads the extension, but this is what's show in the status bar:
![image](https://github.com/SanderRonde/phpstan-vscode/assets/7721989/7a6af683-9131-46cd-b23d-03bea022b1d6)
followed by
![image](https://github.com/SanderRonde/phpstan-vscode/assets/7721989/08cab9a3-23cd-4619-ac7d-451bab6db805)

Output from 1-PHPStan Client.log:
```
[client] Starting extension with configuration: {
	"binPath": "vendor/bin/phpstan",
	"binCommand": [],
	"configFile": "phpstan.neon",
	"rootDir": "",
	"options": [],
	"enableStatusBar": true,
	"memoryLimit": "1G",
	"enabled": true,
	"projectTimeout": 300000,
	"suppressTimeoutMessage": false,
	"paths": {},
	"showProgress": false,
	"enableLanguageServer": true,
	"ignoreErrors": [],
	"suppressWorkspaceMessage": false,
	"pro": false,
	"proTmpDir": ""
}
[client] Initializing done
[client] Showing one-time messages (if needed)
[server] Language server ready
[server] Language server started
[check:1] Check started for project
[status-bar] Showing status bar
[file-watcher] New document active (file:///f%3A/..../src/OCA/OcaViewPeople.php), triggering
[fixer-manager] No file changes, not checking
[file-watcher] New document active (file:///f%3A/..../src/OcaAPI/Controllers/OcaMembershipRecordsController.php), triggering
[fixer-manager] No file changes, not checking
```

This suggests that the extension is trying to do something, but actually it isn't.

After applying a bit of common sense I realise that VSCode has no idea what my dev environment looks like. Changing the `phpstan.binCommand` in settings to use the locally installed PHP binary, e.g. `C:\PHP\php.exe vendor\bin\phpstan` has brought it to life. Hurrah.

**Comment by SanderRonde** [comment]
Thanks for the report and sorry for the slow response. I don't know if I fully understand the problem here so I'll ask for some clarification.
- This doesn't work on the "current" version. That means starting the extension fails right? I'm assuming it's something with the language server crashing is it not? If so I've got a fix for that coming up.
- On the pre-release version startup does work but checking acts like it succeeds but actually doesn't? Is that correct?
- What happens when you run the invalid PHPStan binary in your terminal? Does it throw some big error or does it just do nothing?
